client/signup/steps/domains/index.jsx

```diff
@@ -61,6 +61,25 @@ const DomainsStep = React.createClass( {
 		return { products: productsList.get() };
 	},
 
+	componentWillMount: function() {
+		const { stepName, queryObject } = this.props;
+
+		// skip this step if domain name is given in the url via the `domain` query parameter
+		if ( queryObject && queryObject.domain ) {
```

Comment:
There are a couple issues with this approach:

- `queryObject` always contains the query object that the page loaded with, so the user can't go back to this step without being redirected to the second step in the flow.
- The domains step is shown when the page loads for a couple seconds as the page animates it leaving and the second step appearing.

We have a couple options to handle this:

- Only redirect the user here if this is the first time this step has been shown. This doesn't fix the second issue.
- Remove the `domains` step from this flow entirely and provide the `domainItem` through the `site-or-domain` step. This fixes both issues, but we'll need to link the user back to `/domains` if they want to change their domain selection.

What do you think?